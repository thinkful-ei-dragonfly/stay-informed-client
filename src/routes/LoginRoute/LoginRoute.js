import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'

class LoginRoute extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  render() {
    return (
      <section>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </section>
    );
  }
}

export default LoginRoute
