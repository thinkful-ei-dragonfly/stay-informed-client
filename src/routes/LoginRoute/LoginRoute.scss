$blue: #0063C0;
$red: #E9503E;
.login-wrapper {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  .title,
  .loginForm {
    @media (min-width: 900px) {
      width: 47.5%;
    }
  }

  .title {
    font-size: 2em;
    margin-bottom: 1em;
    @media (min-width: 768px) {
      margin-bottom: 0;
    }
  }

  .loginForm {
    .form-fields {
      display: flex;
      align-items: baseline;
      margin-bottom: 25px;

      label {
        color: $blue;
        width: 33%;
        display: inline-block;
        font-size: 0.85em;
        @media (min-width: 768px) {
          width: 20%;
          font-size: 1em;
        }
      }
      input {
        width: 66%;
        font-size: 1em;

        background: inherit;
        border: 0;
        border-bottom: 2px solid $blue;
        border-style: solid;
        color: $blue;
        @media (min-width: 768px) {
          width: 75%;
          font-size: 1.5em;
        }
      }
    }
    button.submit {
      margin-top: 25px;
      transition: opacity .25s ease;
      opacity: 0.85;
      border: none;
      background: $red;
      color: white;

      &:hover {
        cursor: pointer;
        opacity: 1;
      }
    }
  }

  @media (min-width: 900px) {
    flex-direction: row;
  }
}
