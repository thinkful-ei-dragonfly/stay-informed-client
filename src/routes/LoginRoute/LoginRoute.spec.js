import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginRoute from './LoginRoute'

describe(`LoginRoute component`, () => {
  it('renders a form.LoginRoute by default', () => {
    const wrapper = shallow(<LoginRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
