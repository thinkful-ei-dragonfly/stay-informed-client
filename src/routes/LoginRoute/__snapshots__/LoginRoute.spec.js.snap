// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginRoute component renders a form.LoginRoute by default 1`] = `
<section>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
